### PR TITLE
Refactoring to knockout.js

### PR DESCRIPTION
**Pull request by ilyapoz**
Refactor, review, pull.

### CODE REVIEW

**Comment by ilyapoz** [comment]
@hr0nix this is the js machine to create pain-free interfaces it seems. You can check out the code for wind and landing direction updates. It's quite straightforward.

**Comment by ilyapoz** [comment]
Especially note the code deletions in blueskies.js. More to come.

**Comment by hr0nix** [comment]
Looks way better, I like it.

**Comment by ilyapoz** [comment]
This is a bit more cumbersome than I thought (the refactoring itself, not knockout.js, it's awesome).
TODO:
1. jQuery UI buttonset bindings.
2. map element bindings.

File: scripts/viewmodel.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+function ViewModel() {
```
Comment (ilyapoz) [inline scripts/viewmodel.js:1]: @fib0n, what do you think, is a single ViewModel appropriate for this kind of application or is there a need to divide it into distinct viewmodels?

File: scripts/blueskies.js [hunk 2]
```diff
@@ -482,57 +348,16 @@ function getFullPath(location) {
...
-function updateLandingDirectionValue() {
-    $("#landing-direction-value").html(formatHeading(getLandingDirection()));
-}
-
-function updateLandingPattern() {
-    landingPatternLine.setPath(computeLandingPattern(getCurrentLandingPoint(), getLandingDirection()));
-
-    updateControllabilitySet();
-}
-
```
Comment (ilyapoz) [inline scripts/blueskies.js:535]: Got rid...

File: scripts/blueskies.js [hunk 3]
```diff
@@ -606,167 +416,41 @@ function onShareLinkClick() {
...
-function onShowReachabilitySetCheckboxToggle() {
-    showReachabilitySet = !showReachabilitySet;
-    saveSetting("show-reachability-set", showReachabilitySet);
-
-    updateReachabilitySet();
-}
-
-function onPatternSelect() {
-    setPatternType($(this).attr('id'));
```
Comment (ilyapoz) [inline scripts/blueskies.js:769]: ..of all this...

File: scripts/blueskies.js [hunk 4]
```diff
@@ -937,154 +686,14 @@ function initialize() {
...
-
-    $("#display-ui-element-buttons").buttonset();
-
-    $("#pattern-hide").prop('checked', !showLandingPattern); // We set this before buttonset creation so the buttonset is updated properly
-    $("#pattern-lhs").prop('checked', showLandingPattern && lhsLandingPattern); // We set this before buttonset creation so the buttonset is updated properly
-    $("#pattern-rhs").prop('checked', showLandingPattern && !lhsLandingPattern); // We set this before buttonset creation so the buttonset is updated properly
-    $("#pattern-menu")
-        .buttonset()
-        .children("input")
-            .change(onPatternSelect);
```
Comment (ilyapoz) [inline scripts/blueskies.js:1083]: ...all this bloatware!

File: scripts/viewmodel.js [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+            self.display.controlset,
+            self.display.pattern,
+
+            self.location.id,
+            self.location.custom.coords,
+            self.location.custom.name,
+
+            self.pattern.openingAltitude,
+
+            self.misc.tutorFinished
```
Comment (ilyapoz) [inline scripts/viewmodel.js:236]: All the properties that persist between invocations. Neat imho.

**Comment by ilyapoz** [comment]
Ok, the only things left are `buttonset` bindings, ~~google analytics bindings~~ and testing.

**Comment by ilyapoz** [comment]
Finally, buttonset binding is working.
@hr0nix, please test it a little bit, I must have missed something.
Basically, it should work as before (a week of work!)

**Comment by hr0nix** [comment]
Works perfectly well for me. Great job!

**Comment by ilyapoz** [comment]
Ok, here we go.

**Pull request merged**